libraries/Microsoft.Bot.Builder/TranscriptLoggerMiddleware.cs

```diff
@@ -109,20 +109,13 @@ public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate nextTurn, C
             await nextTurn(cancellationToken).ConfigureAwait(false);
 
             // flush transcript at end of turn
-            var logTasks = new List<Task>();
             while (transcript.Count > 0)
             {
                 // Process the queue and log all the activities in parallel.
                 var activity = transcript.Dequeue();
 
-                // Add the logging task to the list (we don't call await here, we await all the calls together later).
-                logTasks.Add(TryLogActivityAsync(_logger, activity));
-            }
-
-            if (logTasks.Any())
-            {
-                // Wait for all the activities to be logged before continuing.
-                await Task.WhenAll(logTasks).ConfigureAwait(false);
+                // NOTE: We are not awaiting this task by design, TryLogActivity() observes all exceptions and we don't need to or want to block execution on the completion.
+                _ = TryLogActivityAsync(_logger, activity);
```

Comment:
I tested this and this will cause concurrency problems in the FileTranscriptLogger while trying to write all the activities in parelell to the same file, can you change this line by:

```
_ = TryLogActivityAsync(_logger, transcript)
```

And then modify TryLogActivityAsync as follows:

```
        private static async Task TryLogActivityAsync(ITranscriptLogger logger, Queue<IActivity> transcript)
        {
            try
            {
                while (transcript.Count > 0)
                {
                    // Process the queue and log all the activities in parallel.
                    var activity = transcript.Dequeue();
                    await logger.LogActivityAsync(activity).ConfigureAwait(false);
                }
            }
#pragma warning disable CA1031 // Do not catch general exception types (this should probably be addressed later, but for now we just log the error and continue the execution)
            catch (Exception ex)
#pragma warning restore CA2008 // Do not create tasks without passing a TaskScheduler
            {
                Trace.TraceError($"Transcript logActivity failed with {ex}");
            }
        }
```

This will help avoid concurrency problems with the FileTranscriptLogger. It is still possible to get errors if the dev types too fast and the FileTranscriptLogger is still processing but we should fix that in the FileTranscriptLogger by adding a random delay in between retries.

Reply:
Applied the suggested change myself to the branch